habushu-maven-plugin/src/main/java/org/technologybrewery/habushu/migration/CustomPoetrycoreVersionMigration.java

```diff
@@ -0,0 +1,162 @@
+        }
+
+        try {
+            TomlUtils.writeTomlFile(pyProjectTomlFile, fileContent);
+        } catch (IOException e) {
+            throw new BatonException("Problem while writing dependencies to TomlFile during upgrade of poetry-core version!", e);
+        }
+        return true;
+    }
+
+    private boolean IsPoetrycoreUpgradeRequired(Map.Entry<String, Object> dependency) {
+        replacements.clear();
+        String packageName = dependency.getKey();
+
+        if (packageName.equals(TomlUtils.REQUIRES)) {
+            String dependencyVal = dependency.getValue().toString();
+            if (dependencyVal.contains(TomlUtils.POETRY_CORE) && dependencyVal.contains(TomlUtils.EQUALS)) {
+                String poetrycoreVer = dependencyVal.substring(PoetryUtil.getIndexOfFirstDigit(dependencyVal), PoetryUtil.getIndexOfLastDigit(dependencyVal));
+                int endMajorVerIndex = poetrycoreVer.indexOf(TomlUtils.DOT);
+                int poetryCoreMajorVer = Integer.parseInt(poetrycoreVer.substring(0, endMajorVerIndex));
+                int poetryCoreReqMajorVersion =  Integer.parseInt(PoetryUtil.getMajorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT));
+
+                // if the major version is less than required major version then return true.
+                if(poetryCoreMajorVer < poetryCoreReqMajorVersion) {
+                    isMajorVersionUpdateRequired = true;
+                    isMinorVersionUpdateRequired = true;
+                }
+
+                // Now check for the minor version. If the major version is equal to the required
+                // major version but if the minor version is less than the required minor version
+                // then upgrade for minor version is needed.
+                int minorVerIndex = poetrycoreVer.indexOf(TomlUtils.DOT) + 1;
+                int endMinorVerIndex = poetrycoreVer.lastIndexOf(TomlUtils.DOT);
+                int poetryCoreMinorVer = Integer.parseInt(poetrycoreVer.substring(minorVerIndex, endMinorVerIndex));
+                int poetryCoreReqMinorVersion = Integer.parseInt(PoetryUtil.getMinorReqVersion(PoetryUtil.POETRY_CORE_VERSION_REQUIREMENT));
+
+                if ((poetryCoreMajorVer == poetryCoreReqMajorVersion && poetryCoreMinorVer < poetryCoreReqMinorVersion)) {
+                    isMinorVersionUpdateRequired = true;
+                }
+
+                // Return true if major or minor version upgarde is required.
+                if(isMajorVersionUpdateRequired || isMinorVersionUpdateRequired) {
+                    TomlReplacementTuple replacementTuple = new TomlReplacementTuple(packageName, poetrycoreVer, "poetry-core>=1.6.0");
+                    replacements.put(packageName, replacementTuple);
+                    logger.info("Found Poetry-core dependency with version : {} which is less than the required version of 1.6.0 or higher for Habushu. Poetry-core version dependency will be updated to 1.6.0.", poetrycoreVer);
```

Comment:
A: seems like a 4th place this is getting logged - let's keep it tidy, 1 info level log.  We can be more verbose at debug level, if it makes sense.

Reply:
Yes. It should be at the debug level. Made the change to log as debug.